openstack/nova/templates/cell2-conductor-deployment.yaml

```diff
@@ -29,40 +29,44 @@ spec:
         alert-service: nova
 {{ tuple . "nova" "conductor" | include "helm-toolkit.snippets.kubernetes_metadata_labels" | indent 8 }}
       annotations:
-        {{- if .Values.cell2.conductor.config_file.DEFAULT.statsd_enabled }}
+        {{- if or .Values.cell2.conductor.config_file.DEFAULT.statsd_enabled .Values.proxysql.mode }}
         prometheus.io/scrape: "true"
         prometheus.io/targets: {{ required ".Values.alerts.prometheus missing" .Values.alerts.prometheus | quote }}
         {{- end }}
         configmap-etc-hash: {{ include (print $.Template.BasePath "/etc-configmap.yaml") . | sha256sum }}
     spec:
 {{ tuple . "nova" "conductor" | include "kubernetes_pod_anti_affinity" | indent 6 }}
+{{ include "utils.proxysql.pod_settings" . | indent 6 }}
       terminationGracePeriodSeconds: {{ .Values.defaults.default.graceful_shutdown_timeout | add 5 }}
       hostname: nova-conductor
       initContainers:
-        - name: nova-db-sync
-          image: {{ required ".Values.global.registry is missing" .Values.global.registry}}/ubuntu-source-nova-api:{{.Values.imageVersionNovaApi | default .Values.imageVersionNova | default .Values.imageVersion | required "Please set nova.imageVersion or similar"}}
-          command: ['sh', '-c', 'nova-manage db sync --local_cell']
-          volumeMounts:
-            - mountPath: /etc/nova
-              name: etcnova
-            - mountPath: /etc/nova/nova.conf
-              name: nova-etc
-              subPath: nova-cell2.conf
-              readOnly: true
-      containers:
-        - name: nova-conductor
+        - name: dependencies
           image: {{ required ".Values.global.registry is missing" .Values.global.registry}}/ubuntu-source-nova-conductor:{{.Values.imageVersionNovaConductor | default .Values.imageVersionNova | default .Values.imageVersion | required "Please set nova.imageVersion or similar"}}
           imagePullPolicy: IfNotPresent
           command:
-            - dumb-init
             - kubernetes-entrypoint
           env:
             - name: COMMAND
-              value: "nova-conductor --config-file etc/nova/nova.conf --config-file /etc/nova/nova-conductor.conf --config-file /etc/nova/nova-cell2.conf"
+              value: "true"
             - name: NAMESPACE
               value: {{ .Release.Namespace }}
             - name: DEPENDENCY_SERVICE
-              value: "{{ .Release.Name }}-{{ .Values.cell2.name }}-rabbitmq,{{ .Release.Name }}-{{ .Values.cell2.name }}-mariadb"
+              value: "{{ .Release.Name }}-{{ .Values.cell2.name }}-rabbitmq,{{ .Release.Name }}-{{ .Values.cell2.name }}-mariadb"      
+            - name: DEPENDENCY_JOB
+              value: "nova-update-cells2-{{ .Release.Revision }}"
```

Comment:
This means, we cannot restart the pod anymore, if the Job `nova-update-cells2-$rev` got deleted, right? We don't have the same restriction in the non-cell2 conductor deployment. Would it be possible to integrate the cell2 syncing into the main `db-migration` Job?

I'm also wondering why we name this `nova-update-cells2` and not `nova-db-migrate-cell2` as what we're doing here is the `db-migration`'s action for cell2.

Reply:
Now actually [reading the code](https://github.com/sapcc/nova/blob/stable/rocky-m3/nova/cmd/manage.py#L470-L483), I think the `db-migrate` job already updates all cells...
We just have to ensure that the cells are created before the db-migration is run.

Reply:
Then there's not point in having that init-container anymore, is there?

Reply:
I was jumping a bit the gun at reading the help text, and missed that
>  Currently fanout only looks for cell0.

So, back to refactoring that. I'll open a separate PR for that

Reply:
PR is in #3768